### PR TITLE
fix: [M3-7245] - Replaced 'Devices' with 'Services' in Firewall Page

### PR DESCRIPTION
**Pull request by tyler-akamai**
## Description 📝
For the Firewalls, we decided to use the term "Services" instead of "Devices" since CF will eventually expand to LKE, Object Storage, etc.

## Changes  🔄
- "Devices" --> "Services" in 
   - Firewall Landing table 
   - Create Firewall drawer
   - Add Linode to Firewall
   - Add NodeBalancer to Firewall 
- Updated routing so that '/add' opens up the Add NodeBalancer drawer and the Add Linode Drawer

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-10-10 at 12 27 01 PM | Screenshot 2023-10-10 at 11 40 27 AM |
| Screenshot 2023-10-10 at 12 27 16 PM | Screenshot 2023-10-10 at 11 40 47 AM |
| Screenshot 2023-10-10 at 12 27 43 PM | Screenshot 2023-10-10 at 11 41 13 AM |
| Screenshot 2023-10-10 at 12 27 57 PM | Screenshot 2023-10-10 at 11 41 26 AM |

### Prerequisites
(How to setup test environment)
- There are no Prerequisites, you can use either your dev or prod account.

### Reproduction steps
- Checkout the feature/firewall-nodebalancer branch
Firewall Landing Table:
- Navigate to http://localhost:3000/firewalls
- Ensure the 4th column is named "Devices"
Firewall Create Drawer:
- Navigate to http://localhost:3000/firewalls/create
- Ensure the name of the second section is 'Assign devices to the Firewall' and 'devices' is also referred to below the section header
Add Linodes to Firewall Drawer:
- Navigate to http://localhost:3000/firewalls/[firewall_id]/linodes
- Click `Add Linode to Firewall`
- Ensure the word 'Devices' is used in the drawer
Add Linodes to Firewall Drawer:
- Navigate to http://localhost:3000/firewalls/[firewall_id]/nodebalancers
- Click `Add NodeBalancer to Firewall`
- Ensure the word 'Devices' is used in the drawer

### Verification steps 
(How to verify changes)
- Checkout this PR branch
- Ensure "Devices" have been swapped with "Services" in 
   - [ ] Firewall Landing table 
   - [ ] Create Firewall drawer
   - [ ] Add Linode to Firewall
   - [ ] Add NodeBalancer to Firewall 
- [ ] Ensure that http://localhost:3000/firewalls/[firewall_id]/linodes/add opens up the `Add Linode Drawer`
- [ ] Navigate to http://localhost:3000/firewalls/[firewall_id]/linodes and ensure that clicking on `Add Linodes To Firewall` opens the drawer and updates the url by adding '/add'
- [ ] Ensure that http://localhost:3000/firewalls/[firewall_id]/nodebalancers/add opens up the `Add NodeBalancer Drawer`
- [ ] Navigate to http://localhost:3000/firewalls/[firewall_id]/nodebalancers and ensure that clicking on `Add Linodes To Firewall` opens the drawer and updates the url by adding '/add'

## As an Author I have considered 🤔
- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx [hunk 1]
```diff
@@ -41,6 +42,16 @@ export const FirewallDeviceLanding = React.memo(
 
     const theme = useTheme();
 
+    const history = useHistory();
+    const routeMatch = useRouteMatch();
+    const location = useLocation();
+
+    React.useEffect(() => {
```
Comment (tyler-akamai) [inline packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx:49]: This is to check if the url ends with a '/add' and if so, it opens up the drawer.

File: packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx [hunk 2]
```diff
@@ -29,7 +30,7 @@ export const formattedTypes = {
 };
 
 const helperText =
-  'Assign one or more devices to this firewall. You can add devices later if you want to customize your rules first.';
+  'Assign one or more Services to this Firewall. You can add Services later if you want to customize your rules first.';
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx:33]: Just checking if we want to capitalize this everywhere?

File: packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx:49]: Let's add a unit test for this scenario where we might want to redirect someone to this drawer, look at https://github.com/linode/manager/blob/develop/packages/manager/src/features/TopMenu/AddNewMenu/AddNewMenu.test.tsx#L51 for an example

**Review by jaalah-akamai — APPROVED** [review]
Pending unit test.
- [x] Confirmed the `/add` opens the drawer for both linodes and nodebalancers
- [x] Confirmed `devices` is now `services`
- [x] Code looks clean. 🧼

**Review by jdamore-linode — APPROVED** [review]
Looks good to me! Approved pending confirmation on `Services` casing.

**Comment by tyler-akamai** [comment]
> Looks good to me! Approved pending confirmation on `Services` casing.

File: packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx [hunk 3]
```diff
@@ -29,7 +30,7 @@ export const formattedTypes = {
 };
 
 const helperText =
-  'Assign one or more devices to this firewall. You can add devices later if you want to customize your rules first.';
+  'Assign one or more services to this firewall. You can add services later if you want to customize your rules first.';
```
Comment (tyler-akamai) [inline packages/manager/src/features/Firewalls/FirewallDetail/Devices/FirewallDeviceLanding.tsx:33]: Confirmed with Kendall: @jdamore-linode @jaalah-akamai 
- NodeBalancers and Linodes are product names, and should be capitalized.
- Services is not a product name and Firewalls is a generic product name, therefore should be lowercase
- Services and Firewalls should be uppercase in a header/title but not in a sentence

**Review by jaalah-akamai — APPROVED** [review]

**Pull request merged**